File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  user_id: string;
  type: 'TALENT_ALERT' | 'HIDDEN_SKILL' | 'ROLE_MATCH' | 'SKILL_GAP';
  title: string;
  message: string;
  is_read: boolean;
  created_at: string;
}

File: backend/app/models/notification.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Text, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.db.base import Base


class Notification(Base):
    __tablename__ = "notifications"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    type = Column(String(50), nullable=False, index=True)  # TALENT_ALERT, HIDDEN_SKILL, ROLE_MATCH, SKILL_GAP
    title = Column(String(255), nullable=False)
    message = Column(Text, nullable=False)
    is_read = Column(Boolean, default=False, nullable=False, index=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), nullable=False)

    # Relationships
    user = relationship("User", back_populates="notifications")
